### DISCUSSION TITLE
How to generate class file from Jimple file?

### DISCUSSION THREAD

**Discussion by JialeTomTian** [soot-oss/SootUp#765] [Q&A] [answered]
I understand using `SootClass.print()` to get the Jimple representation of a class. But how could I convert a Jimple file to a class file in Sootup.

**Comment by kadirayk** [comment, marked as answer]
This is not possible at the moment. But it would be nice to have a bytecode backend.

**Reply by imcsq** [reply]
Will it be possible soon? I see a task in your todo list to support outputting the class file. The feature is wanted in order to realize the usefulness of body interceptors & class mutation.

I tried the alternative solution of running the printed jimple file with the soot, but it failed. Many features in the jimple files generated with sootup (latest develop version so far) cannot be handled by the soot jimpler parser, e.g., the [SUPER] class modifier and [varargs] method modifier. Sometimes I also get the error "No method source set for method <...mymethod...>".

Thanks!

**Reply by swissiety** [reply]
Generating binaries from Jimple is not a planned for the near future on our side. But you are welcome to contribute :)
(Sidenote: a BodyInterceptors purpose is not necessarily optimizing execution at runtime but already improving the generated Jimple from Bytecode)

**Reply by imcsq** [reply]
Thanks for the quick reply. Then is it possible to execute the jimple code? Or is there a method to perform profiling/instrumenting with sootup?

**Comment by JialeTomTian** [comment]
Hi, to follow up. In that case, if I have a Jimple file. Is there a way of going about loading that Jimple file and executing it at the moment?

**Reply by kadirayk** [reply]
This is also not possible at the moment.

**Reply by swissiety** [reply]
If it helps there exists a repo of a [Jimple interpreter](https://github.com/secure-software-engineering/Jimple-Interpreter) from a hackathon(?) but its not maintained and I'm not sure how much functionality is implemented.
